app/controllers/tasks_controller.rb

```diff
@@ -13,35 +13,88 @@ class TasksController < ApplicationController
   end
 
   post '/tasks' do
-    task = Task.new(description: params[:description])
+    task = Task.new(description: params[:description], user: current_user)
 
     if task.save
-      redirect "/"
+      if is_json_request?
+        status 201
+        json task
+      else
+        redirect "/"
+      end
     else
-      flash.now[:errors] = task.errors.full_messages.join("; ")
-      erb :"tasks/new.html"
+      if is_json_request?
+        status 400
+        json errors: task.errors.full_messages
+      else
+        flash.now[:errors] = task.errors.full_messages.join("; ")
+        erb :"tasks/new.html"
+      end
     end
   end
 
   get '/tasks/:id' do
     task = Task.find(params[:id])
-    erb :"tasks/edit.html", locals: { task: task }
+    if is_json_request?
+      if task.user == current_user
+        status 200
+        json task
+      else
+        status 404
+        json({})
+      end
+    else
+      erb :"tasks/edit.html", locals: { task: task }
+    end
+  rescue ActiveRecord::RecordNotFound
+    status 404
+    json({})
   end
 
   put '/tasks/:id' do
     task = Task.find(params[:id])
```

Comment:
If you use `current_user.tasks.find_by_id(params[:id])` here you will prevent `ActiveRecord::RecordNotFound` errors as well as preventing unauthorized users from accessing a task by id.